src/main/kotlin/calculator/StringCalculator.kt

```diff
@@ -0,0 +1,46 @@
+package calculator
+
+data class StringCalculator(val totalNumber: Int) {
+
+    companion object {
+        fun calculate(input: String?): Int {
+            if (input.isNullOrBlank()) {
+                return 0
+            }
+
+            if (input.length == 1 && input.all { it.isDigit() }) {
+                return input.toInt()
+            }
+
+            val (delimiter, text) = parseDelimiterAndText(input)
+            validate(text, delimiter)
+            return splitString(text, delimiter)
+        }
+
+        private fun parseDelimiterAndText(input: String?): Pair<String, String?> {
+            var delimiter = "[,:]"
+            var text = input
+            val result = text?.let { Regex("//(.)\n(.*)").find(it) }
+            result?.let {
+                delimiter = it.groupValues[1]
+                text = it.groupValues[2]
+            }
+            return Pair(delimiter, text)
```

Comment:
Pair객체를 만들고 싶다면 `to` infix 함수를 활용할 수도 있어요! 😃

```kotlin
public infix fun <A, B> A.to(that: B): Pair<A, B> = Pair(this, that)
```

```suggestion
            return delimiter to text
```